internal/services/redis/redis_cache_resource.go

```diff
@@ -935,29 +935,31 @@ func expandRedisConfiguration(d *pluginsdk.ResourceData) (*redis.RedisCommonProp
 	skuName := d.Get("sku_name").(string)
 
 	if v := raw["maxclients"].(int); v > 0 {
-		output.Maxclients = utils.String(strconv.Itoa(v))
+		output.Maxclients = pointer.To(strconv.Itoa(v))
 	}
 
 	if d.Get("sku_name").(string) != string(redis.SkuNameBasic) {
 		if v := raw["maxmemory_delta"].(int); v > 0 {
-			output.MaxmemoryDelta = utils.String(strconv.Itoa(v))
+			output.MaxmemoryDelta = pointer.To(strconv.Itoa(v))
 		}
 
 		if v := raw["maxmemory_reserved"].(int); v > 0 {
-			output.MaxmemoryReserved = utils.String(strconv.Itoa(v))
+			output.MaxmemoryReserved = pointer.To(strconv.Itoa(v))
 		}
 
 		if v := raw["maxfragmentationmemory_reserved"].(int); v > 0 {
-			output.MaxfragmentationmemoryReserved = utils.String(strconv.Itoa(v))
+			output.MaxfragmentationmemoryReserved = pointer.To(strconv.Itoa(v))
 		}
 	}
 
 	if v := raw["maxmemory_policy"].(string); v != "" {
-		output.MaxmemoryPolicy = utils.String(v)
+		output.MaxmemoryPolicy = pointer.To(v)
 	}
 
 	if v := raw["data_persistence_authentication_method"].(string); v != "" {
-		output.PreferredDataPersistenceAuthMethod = utils.String(v)
+		output.PreferredDataPersistenceAuthMethod = pointer.To(v)
+	} else {
+		output.PreferredDataPersistenceAuthMethod = pointer.To("")
```

Comment:
Would it make sense to replace this with the below since we are now setting it whether or not the string is empty?

`output.PreferredDataPersistenceAuthMethod = pointer.To(raw["data_persistence_authentication_method"].(string))`

Reply:
@catriona Empty string will be ignore but `null` will be accepted by service

Reply:
I believe this should still be ok as it won't be sent in the payload if it is null

Reply:
Updated